internal/istio/virtualservice_test.go

```diff
@@ -249,18 +249,18 @@ func createScheme(t *testing.T) *machineryruntime.Scheme {
 }
 
 func createGateways() *istioclientapiv1beta1.GatewayList {
-	gateways := []*istioclientapiv1beta1.Gateway{}
-	for gatewayIndex := 0; gatewayIndex < 3; gatewayIndex++ {
+	gateways := make([]*istioclientapiv1beta1.Gateway, 0, 3)
+	for gatewayIndex := range 3 {
```

Comment:
I would define a variable for this `3`, please also check if the amount of server and hosts id should be same as gateway (always 3?), I'm a bit doubt if the amount of them should be same.